### ISSUE TITLE
AttributeError: 'WindowsPath' object has no attribute 'with_stem'

### ISSUE THREAD

**Issue by CrystalWindSnake** [zauberzeug/nicegui#2319]
### Description

error message

```
  File "D:\ngProject\ng-venv1\lib\site-packages\nicegui\storage.py", line 97, in __init__
    self.migrate_to_utf8()
  File "D:\ngProject\ng-venv1\lib\site-packages\nicegui\storage.py", line 158, in migrate_to_utf8
    new_filepath = filepath.with_stem(filepath.stem.replace('_', '-'))
AttributeError: 'WindowsPath' object has no attribute 'with_stem'

Process finished with exit code 1
```

The `with_stem` method is new in python 3.9.

**Comment by rodja** [comment]
Ok. How would we need to code this for Python 3.8 compatibility? Would you like to create a pull request?

**Comment by falkoschindler** [comment]
I'd suggest to simply changing the name instead of the stem. This shouldn't make a difference, since the suffix ".json" doesn't contain "-" or "_" anyway. And `with_name` is available in Python 3.8.

So we can simply replace
```py
new_filepath = filepath.with_stem(filepath.stem.replace('_', '-'))
```
with
```py
new_filepath = filepath.with_name(filepath.name.replace('_', '-'))
```

@CrystalWindSnake Let us know if you want to post a PR or if we should fix it ourselves. We're fine with both. 🙂

**Comment by CrystalWindSnake** [comment]
> I'd suggest to simply change the name instead of the stem. This shouldn't make a difference, since the suffix ".json" doesn't contain "-" or "_" anyway. And `with_name` is available in Python 3.8.
> 
> So we can simply replace
> 
> ```python
> new_filepath = filepath.with_stem(filepath.stem.replace('_', '-'))
> ```
> 
> with
> 
> ```python
> new_filepath = filepath.with_name(filepath.name.replace('_', '-'))
> ```
> 
> @CrystalWindSnake Let us know if you want to post a PR or if we should fix it ourselves. We're fine with both. 🙂

I will gladly create a pull request
